### PR TITLE
Migration of gst-launch testcases to SSAT @open sesame 11/02 17:37

### PR DESCRIPTION
**Pull request by myungjoo**
We are going to use ssat for gst-launch-based test cases
( https://github.com/myungjoo/SSAT )

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

TODO before Merging:
- Work on other test cases ( a few directories are not migrated, yet)
- Enable SSAT in TAOS-CI @ Tizen (Ubuntu-PPA is ready)

### CODE REVIEW

**Comment by myungjoo** [comment]
Need to add "ssat.rpm" for TAOS-CI @ohsewon @leemgs

File: tests/nnstreamer_converter/runTest.sh [hunk 1]
```diff
@@ -1,5 +1,32 @@
 #!/usr/bin/env bash
-source ../testAPI.sh
+
```
Comment (leemgs) [inline tests/nnstreamer_converter/runTest.sh]: Could you write a doxygen tag  in this PR  to generate doxygen document for a shell script as well as  *.c source code?  I think that now is appropriate time to append doxygen tags becasue the  ssat scripts are stable compared to the existing facility. :) 

* Reference: 
https://github.com/nnsuite/TAOS-CI/blob/master/ci/doc/doxygen-documentation.md#case-study-bash

File: tests/nnstreamer_converter/runTest.sh [hunk 2]
```diff
@@ -12,18 +39,22 @@ else
...
+gstTest "videotestsrc num-buffers=1 ! video/x-raw,format=RGB,width=280,height=40,framerate=0/1 ! tee name=t ! queue ! videoconvert ! video/x-raw, format=BGRx ! tensor_converter silent=TRUE ! filesink location=\"test.bgrx.log\" sync=true t. ! queue ! filesink location=\"test.rgb.log\" sync=true" 1R 0 0 $PERFORMANCE
 
-gstTest "--gst-plugin-path=${PATH_TO_PLUGIN} videotestsrc num-buffers=1 ! video/x-raw,format=GRAY8,width=280,height=40,framerate=0/1 ! queue ! tensor_converter silent=TRUE ! filesink location=\"test.gray8.log\" sync=true" 1
+gstTest "videotestsrc num-buffers=1 ! video/x-raw,format=GRAY8,width=280,height=40,framerate=0/1 ! queue ! tensor_converter silent=TRUE ! filesink location=\"test.gray8.log\" sync=true" 1G 0 0 $PERFORMANCE
 
-compareAllSizeLimit testcase01.bgrx.golden test.bgrx.log 1-1
-compareAllSizeLimit testcase01.rgb.golden test.rgb.log 1-2
-compareAllSizeLimit testcase01.gray8.golden test.gray8.log 1-3
+callCompareTest testcase01.bgrx.golden test.bgrx.log 1-1 "BGRX Golden Test" 1 0
```
Comment (leemgs) [inline tests/nnstreamer_converter/runTest.sh]: Where can we find  a official document to use the public APIS such as callCompareTest(3)?

File: tests/nnstreamer_converter/runTest.sh (hunk 1 above)
Comment (myungjoo) [reply tests/nnstreamer_converter/runTest.sh]: Done. Applied the standard SSAT template.

File: tests/nnstreamer_converter/runTest.sh [hunk 3]
```diff
@@ -12,18 +29,22 @@ else
...
+callCompareTest testcase01.rgb.golden test.rgb.log 1-2 "RGB Golden Test" 1 0
+callCompareTest testcase01.gray8.golden test.gray8.log 1-3 "Gray8 Golden Test" 1 0
 
 function do_test {
-	gstTest "--gst-plugin-path=${PATH_TO_PLUGIN} filesrc location=testcase02_${1}_${2}x${3}.png ! pngdec ! videoscale ! imagefreeze ! videoconvert ! video/x-raw,format=${1},width=${2},height=${3},framerate=0/1 ! tensor_converter silent=TRUE ! filesink location=\"testcase02_${1}_${2}x${3}.log\" sync=true" ${4}
+	gstTest "--gst-plugin-path=${PATH_TO_PLUGIN} filesrc location=testcase02_${1}_${2}x${3}.png ! pngdec ! videoscale ! imagefreeze ! videoconvert ! video/x-raw,format=${1},width=${2},height=${3},framerate=0/1 ! tensor_converter silent=TRUE ! filesink location=\"testcase02_${1}_${2}x${3}.log\" sync=true" ${4} 0 0 $PERFORMANCE
```
[the remark this answers is not in the document]
Comment (myungjoo) [reply tests/nnstreamer_converter/runTest.sh]: Added

File: tests/nnstreamer_converter/runTest.sh (hunk 2 above)
Comment (myungjoo) [reply tests/nnstreamer_converter/runTest.sh]: https://raw.githubusercontent.com/wiki/myungjoo/SSAT/doxygen/refman.pdf  (SSAT Wiki)

File: packaging/nnstreamer.spec [hunk 4]
```diff
@@ -46,6 +46,9 @@ BuildRequires: lcov
 # BuildRequires:	taos-ci-unittest-coverage-assessment
 %endif
 
+# Unit Testing Uses SSAT (hhtps://github.com/myungjoo/SSAT.git)
+BuildRequires: ssat
```
[the remark this answers is not in the document]
Comment (leemgs) [reply packaging/nnstreamer.spec]: * ssat tizen packages - https://build.tizen.org/package/show/devel:AIC:Tizen:5.0:nnsuite/ssat

**Comment by myungjoo** [comment]
Ok. Let's go!

**Pull request merged**